src/vuex/modules/session.js

```diff
@@ -148,16 +144,23 @@ export default () => {
 
       state.externals.track(`event`, `session`, `sign-in`, sessionType)
     },
-    signOut({ state, commit, dispatch }) {
+    signOut({ state, commit, dispatch }, networkId) {
       state.externals.track(`event`, `session`, `sign-out`)
 
-      dispatch(`resetSessionData`)
+      dispatch(`resetSessionData`, networkId)
       commit(`setSignIn`, false)
     },
-    resetSessionData({ commit, state }) {
+    resetSessionData({ commit, state }, networkId) {
       state.history = ["/"]
       commit(`setUserAddress`, null)
-      localStorage.removeItem(`session`)
+      // Hack. In case we receive the networkId parameter, we reset session data for
```

Comment:
In which case wouldn't we receive the networkId?

Reply:
When called from `connection.js` (while executing `setNetwork`). This is the hack

Reply:
is `session` in localStorage set at any point?

Reply:
I guess during signin

Reply:
Wrooooong ;P

Reply:
:crying_cat_face: